sirepo/package_data/static/js/elegant.js

```diff
@@ -1443,3 +1443,72 @@ SIREPO.app.directive('srBunchEditor', function(appState, panelState) {
         },
     };
 });
+
+
+SIREPO.app.directive('viewLogIframe', function(appState, requestSender) {
+    return {
+        restrict: 'A',
+        scope: {},
+        template: `
+            <a href data-ng-click="viewLog()">View Log</a>
+            <div class="modal fade" id="sr-iframe-text-view" tabindex="-1" role="dialog">
+              <div class="modal-dialog modal-lg">
+                <div class="modal-content">
+                  <div class="modal-header bg-warning">
+                    <span class="lead modal-title text-info">Log</span>
+                    <div class="sr-panel-options pull-right">
+                      <a data-ng-href="{{ downloadLog() }}" target="_blank">
+                        <span class="sr-panel-heading glyphicon glyphicon-cloud-download"></span>
+                      </a>
+                      <button type="button" class="close" data-dismiss="modal" style="margin-left: 10px">
+                        <span>&times;</span>
+                      </button>
+                    </div>
+                  </div>
+                  <div class="modal-body" style="padding: 0">
+                    <iframe id="sr-text-iframe"
+                      style="border: 0; width: 100%; height: 80vh" src=""></iframe>
+                  </div>
+                </div>
+              </div>
+            </div>
+        `,
+        controller: function(appState, elegantService, requestSender, $scope) {
+
+            function setIFrameHTML(html) {
+                $('#sr-text-iframe').contents().find('html').html(html);
```

Comment:
I think this can be simplified to `$('#sr-text-iframe').attr("srcdoc", html);`. srcdoc is the attribute for setting an html string on an iframe https://www.w3schools.com/tags/att_iframe_srcdoc.asp